### ISSUE TITLE
Control node is instance creation is not idempotent

### ISSUE THREAD

**Issue by termlen0** [ansible/workshops#95]
Yesterday I ran into an issue where I had to rerun the provisioner with an updated student count. This caused 2 control nodes to be created for previously created student pods. The routers were still idempotent.

**Comment by IPvSean** [comment]
yeah this is known issue that I should have tracked here, tagging @jmcleroy so he knows we are working on it

**Comment by jmcleroy** [comment]
Tried removing the exact count attribute from the instances to see if that would resolve the issue without success. Still not seeing why the tagging it getting off as the nodes are being retagged as student1 for instance when you add a second student and so forth.

**Comment by termlen0** [comment]
The reason for the issue is due to the  use of the index in the subsequent tagging task. That will always start at 0 when we re-run the provisioner causing duplicates.

**Comment by IPvSean** [comment]
MORE INFO: so the problem description is if you use `student_count: 10` then later you want to add two students and do `student_count: 12` with the name workshop name it will not add student 11 and student 12 it adds more instances to student1 and student2, we need to use a filter to order the instances when we grab facts

**Comment by rlopez133** [comment]
@IPvSean can we close this? its from 2018.

**Comment by IPvSean** [comment]
yes please
